buckets/local/watch.go

```diff
@@ -0,0 +1,131 @@
+package local
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/radovskyb/watcher"
+	"github.com/textileio/go-threads/api/client"
+	"github.com/textileio/textile/buckets"
+)
+
+const defaultInterval = time.Millisecond * 100
+
+// Watch watches for and auto-pushes local bucket changes at an interval,
+// and listens for and auto-pulls remote changes as they arrive.
+func (b *Bucket) Watch(ctx context.Context, opts ...WatchOption) error {
```

Comment:
Watch logic is here... slightly tricky one. The goal is to push local changes when they happen, but to avoid doing a pull _when a push is happening_. Pulls are kicked off by remote events, which will be fired every time a file is added (could be many during a push). 

This method should error out when either of the listeners fails.